src/utils/requests/UserRequests.test.js

```diff
@@ -0,0 +1,105 @@
+import { login, register } from "./UserRequests";
+import sinonStubPromise from "sinon-stub-promise";
+import sinon from "sinon"
+import assert from "assert";
+
+const loginInput = {
+    "email":"test@email.com",
+    "password":"password",
+};
+
+const registerInput = {
+    "email":"test@email.com",
+    "password":"password",
+    "firstName":"test",
+    "lastName":"test",
+}
+
+const goodResponse = {
+    "status": 200,
+}
+
+const badResponse = {
+    "status": 400,
+    "timestamp": "a date",
+    "message": "error message from server",
+    "errors": [
+      {
+        "field": "string",
+        "rejectedValue": {},
+        "message": "string",
+      }
+    ]
+}
+
+sinonStubPromise(sinon)
+
+describe("UserRequests", () => {
+    describe("login", () => {
+
+        let mockObj;
+
+        beforeEach(() => {
+            mockObj = sinon.stub(window, "fetch");
+        });
+
+        afterEach(() => {
+            mockObj.restore();
+        });
+
+        it("Should login successfully", async () => {
+            window.fetch.returns(mockApiResponse(goodResponse));
+            let res = await register(loginInput);
+            
+            assert.equal(res.status, goodResponse.status);
+        });
+
+        it("Should return error from unsuccesful post when logging", async () => {
+            window.fetch.returns(mockApiResponse(badResponse));
+
+            let res = await login(loginInput);
+            
+            assert.equal(res instanceof Error, true);
+            assert.equal(res.message, badResponse.message);
```

Comment:
if you're checking if your mock fails, then you can use assert's [throw](https://nodejs.org/api/assert.html#assert_assert_throws_fn_error_message) function